plugins/newsdownloader.koplugin/main.lua

```diff
@@ -316,12 +316,20 @@ function NewsDownloader:setCustomDownloadDirectory()
     }:chooseDir()
 end
 
+local function getAbsoluteDirPath(path)
+    if string.sub(path, 1, 1) == "." then
+        return lfs.currentdir() .. string.sub(path, 2)
```

Comment:
`lfs.currentdir()` is *not* (necessarily) pointing to the same folder as `DataStorage:getDataDir()`. This logic shouldn't be taken outside of the dedicated module. If we decide to keep this function it should be replaced by `DataStorage:getFullDataDir()`. But what is it you're missing exactly? It might be something worth adding to DataStorage.

If I understand correctly, you're looking for something along these lines:

```lua
DataStorage:getFullDataSubDir(sub_dir)
    return string.format("%s%s", self:getFullDataDir(), sub_dir)
```